config/deployments/flagd/deployment.yaml

```diff
@@ -0,0 +1,44 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: flagd
+  name: flagd
+spec:
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+  selector:
+    matchLabels:
+      app: flagd
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: flagd
+        app: flagd
+    spec:
+      containers:
+      - name: flagd
+        image: ${IMG}
+        readinessProbe:
+          httpGet:
+            path: /readyz
+            port: 8014
+          initialDelaySeconds: 5
+          periodSeconds: 5
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 8014
+          initialDelaySeconds: 5
+          periodSeconds: 60
+        ports:
+          - containerPort: 8013
+        args:
+          - start
+          - --uri
+          - https://raw.githubusercontent.com/open-feature/flagd/main/samples/example_flags.flagd.json
```

Comment:
Nit - shall we avoid this reference to flags and replace it with volume mounted file path reference [1]

ex:

```
          args: [ "start", "--uri", "file:/etc/flagd/config.json" ]
          volumeMounts:
            - name: config-volume
              mountPath: /etc/flagd
      volumes:
        - name: config-volume
          configMap:
            name: open-feature-flags
            items:
              - key: flags
                path: config.json
                
---

# ConfigMap for Flagd OpenFeatuer provider
apiVersion: v1
kind: ConfigMap
metadata:
  name: open-feature-flags
data:
  flags: |
{{Flag JSON }}
```

The reason is to decouple tests with a resource living in the main. Given that test's evaluation rely on flag content, we should avoid circular dependency on repo main 

[1] - https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md#filepath-provider

Reply:
Also a posibility, will adapt!